SPARCbot.py

```diff
@@ -168,8 +184,276 @@ async def iwantto(self, ctx, role: str):
+        bet_rows = []
+
+        try:
+            #give an update
+            await timed_send(ctx,'Viewing latest '+str(view)+' bets ' + ('with status `' + status +'`' if status else ''))
+            while len(bet_rows) < view and current_bet_id > 0:
+                bet_key = 'bet_'+str(current_bet_id)
+
+                #find bets that match the criteria
+                if bet_key in bet_log:
+                    if config['bet_status'][bet_log[bet_key]['status']] == status or not status:
+                        bet_row = []
+
+                        #add rows to bet log
+                        for cname in config['bet_log_columns']:
+                            #for when sellers don't exist
+                            if cname not in bet_log[bet_key]:
+                                val = "N/A"
+                            #truncate strings that are too long
+                            elif isinstance(bet_log[bet_key][cname],str):
+                                if cname == 'status':
+                                    val = config['bet_status'][bet_log[bet_key][cname]]
+                                else:
+                                    val = bet_log[bet_key][cname][:35] + (bet_log[bet_key][cname][35:] and '...')
+                            #return nickname or id if not a nickname
+                            else:
+                                val = await get_nick_from_id(ctx,bet_log[bet_key][cname])
+                            bet_row = bet_row+[val]
+                        bet_rows.append(bet_row)
+
+                current_bet_id = current_bet_id - 1
+            await timed_send(ctx, '```'+tabulate(bet_rows,headers=config['bet_log_columns'])+'```')
+        except Exception as e:
+            await timed_send(ctx, 'Hmm...that didn\'t seem to work.')
+            raise
+
+    @commands.command()
+    async def view(self,ctx,bet_id):
+        '''<bet_id>: view the terms for a single bet'''
+        #load bets
+        try:
+            with open("bet_log.json", "r") as read_file:
+                bet_log = json.load(read_file)
+            bet_key = 'bet_'+str(bet_id)
+            message = '>>> '
+
+            for cname in config['bet_log_columns']:
+                #for when sellers don't exist
+                if cname not in bet_log[bet_key]:
+                    val = "N/A"
+                #truncate strings that are too long
+                elif isinstance(bet_log[bet_key][cname],str):
+                    val = bet_log[bet_key][cname]
+                #return nickname or id if not a nickname
+                else:
+                    val = await get_nick_from_id(ctx,bet_log[bet_key][cname])
+                message = message + '**'+cname+'**: '+val+'\n'
+            await timed_send(ctx,message)
+        except KeyError as e:
+            await timed_send(ctx,'That bet doesn\'t exist!')
+            raise
+        except Exception:
+            await timed_send(ctx, 'Hmm...that didn\'t seem to work.')
```

Comment:
more informative please :P (it can be helpful to just raise the exception and have it printed in the console where the bot is running)

Reply:
and the error messages themselves can be configurable in the config file, if you want